docs/csharp/whats-new/csharp-8.md

```diff
@@ -0,0 +1,314 @@
+
+```csharp
+public class Point
+{
+    public int X { get; }
+    public int Y { get; }
+
+    public Point(int x, int y) => (X, Y) = (x, y);
+
+    public void Deconstruct(out int x, out int y) =>
+        (x, y) = (X, Y);
+}
+```
+
+The following method uses the **positional pattern** to extract the values of `x` and `y`. Then, it uses a `when` clause to determine the quadrant of the point:
+
+```csharp
+static string Quadrant(Point p) => p switch
+{
+    (0, 0) => "origin",
+    (var x, var y) when x > 0 && y > 0 => "Quadrant 1",
+    (var x, var y) when x < 0 && y > 0 => "Quadrant 2",
+    (var x, var y) when x < 0 && y < 0 => "Quadrant 3",
+    (var x, var y) when x > 0 && y < 0 => "Quadrant 4",
+    (var x, var y) => "on a border",
+    _ => "unknown"
+};
+```
+
+## using declarations
+
+A **using declaration** is a variable declaration preceded by the `using` keyword. It tells the compiler that the variable being declared should be disposed at the end of the enclosing scope. For example, consider the following code that writes a text file:
+
+```csharp
+static void WriteLinesToFile(IEnumerable<string> lines)
+{
+    using (var file = new System.IO.StreamWriter("WriteLines2.txt"))
+    {
+        foreach (string line in lines)
+        {
+            // If the line doesn't contain the word 'Second', write the line to the file.
+            if (!line.Contains("Second"))
+            {
+                file.WriteLine(line);
+            }
+        }
+    } // file is disposed here
+}
+```
+
+A few braces are removed by replacing the `using` statement with a `using` declaration:
+
+```csharp
+static void WriteLinesToFile(IEnumerable<string> lines)
+{
+    using var file = new System.IO.StreamWriter("WriteLines2.txt");
+    foreach (string line in lines)
+    {
+        // If the line doesn't contain the word 'Second', write the line to the file.
+        if (!line.Contains("Second"))
+        {
+            file.WriteLine(line);
+        }
+    }
+// file is disposed here
+}
+```
+
+Using declarations provide the same behavior when exceptions are thrown: the `Dispose()` call is in a `finally` block.s
+
+## Static local functions
+
+You can now add the `static` modifier to local functions to ensure that local function doesn't capture (reference) any variables from the enclosing scope. Doing so generates `CS8421`, "A static local function can't contain a reference to <variable>." 
+
+Consider the following code. The local function `LocalFunction` accesses the variable `y`, declared in the enclosing scope (the method `M`). Therefore, `LocalFunction` can't be declared with the `static` modifier:
```

Comment:
I agree with this comment. One place I used a local function was to do a comparison on a backing `List`. I had a `AddObject` type of method that looked at the object and added it to one of four lists and then called `List.Sort` passing in the local function to do the compare. In C#8 that will now be a static function since it was a compare method

```csharp
int CompareComponent(IComponent left, IComponent right)
{
    if (left.SortOrder > right.SortOrder)
        return 1;

    if (left.SortOrder < right.SortOrder)
        return -1;

    return 0;
}
```

Reply:
I went with a simpler example for this topic. When I write the "explore" topic, with interactive samples, this is a good one.